app/models/report.rb

```diff
@@ -4,14 +4,50 @@ class Report < ApplicationRecord
   belongs_to :user
   has_many :comments, as: :commentable, dependent: :destroy
 
+  has_many :active_relationships, class_name: 'Relationship', foreign_key: :mentioning_id, inverse_of: 'mentioning', dependent: :destroy
+  has_many :mentioning_reports, through: :active_relationships, source: :mentioned
+
+  has_many :passive_relationships, class_name: 'Relationship', foreign_key: :mentioned_id, inverse_of: 'mentioned', dependent: :destroy
+  has_many :mentioned_reports, through: :passive_relationships, source: :mentioning
+
   validates :title, presence: true
   validates :content, presence: true
 
+  after_create :create_mentions
+  after_update :update_mentions
+
   def editable?(target_user)
     user == target_user
   end
 
   def created_on
     created_at.to_date
   end
+
+  def create_mentions
+    mentioned_ids = find_mentioned_ids(content)
```

Comment:
contentはselfが持っている値なので引数で渡さなくてよいかも。

```suggestion
    mentioned_ids = find_mentioned_ids_in_content
```